### DISCUSSION TITLE
batch search queries

### DISCUSSION THREAD

**Discussion by arita37** [qdrant/qdrant#184] [Q&A]
Currently,
 one can query one vector a time the topk vector.

Do you think it is possible to have a batch search queries,
something like

search_batch(
vectors  :  list of numpy vectors.
fliter   :  filtering criteria, same for all input vectors.

return : 

list of results   : 
   res[0] : result for input vector index 0,
   res[1] : result for input vector index 1,
   res[2] : result for input  vector index 2,

Technically, we can do it on client python side (multi threading),
but it might be more optimized to do it on server side (ie sharing the filtering criteria).

Standard KVS, RocksDB, Cassandra are doing this...

thanks for your good DB.

**Comment by generall** [comment]
Hi @arita37, it might be a useful feature, indeed! Thanks for your proposal

**Comment by arita37** [comment]
Hello,
 Additionnally, we have this use case :
 
 1 vector numpy.
 mutiple filtering with OR operator:
    filter_cond1   OR filter_cond2 OR filter_cond3
 
 results : 
       filter_cond1  --> topk list
       filter_cond2  --> topk list
       filter_cond3  --> topk list
 
 Typical use case is , when each vector is associated with category_id
      vector, cat_id
      
  We retrieve the top-k for cat_id in (2,5,7,9).
 
  This is general use case and having the compute in batch inside the server
  may be much faster...  Mutiple queries for same vector takes extra time...
